ENIGMAsystem/SHELL/Universal_System/roomsystem.cpp

```diff
@@ -580,45 +578,31 @@ int room_tile_clear(int indx)
 {
   errcheck(indx,"Nonexistent room", 0);
   enigma::roomstruct *rm = enigma::roomdata[indx];
-  enigma::tile *newtiles = new enigma::tile[1];
-  rm->tilecount = 0;
 
-  if (enigma::tile_alter) delete[] rm->tiles;
-  rm->tiles = newtiles;
+  rm->tiles.clear();
   enigma::tile_alter = true;
   return 1;
 }
 
 int room_instance_add(int indx, int x, int y, int obj)
-  {
+{
   errcheck(indx,"Nonexistent room", 0);
   enigma::roomstruct *rm = enigma::roomdata[indx];
-  const int icount = rm->instancecount++;
-  enigma::inst *in = rm->instances;
-  enigma::inst *newinst = new enigma::inst[icount + 1];
-  for (int insti = 0; insti < icount; insti++)
-    newinst[insti] = in[insti];
-
-  newinst[icount].id = enigma::maxid++;
-  newinst[icount].x = x;
-  newinst[icount].y = y;
-  newinst[icount].obj = obj;
-
-  if (enigma::instance_alter) delete[] rm->instances;
-    rm->instances = newinst;
-    enigma::instance_alter = true;
-    return 1;
-  }
+  rm->instances.emplace_back(
+    enigma::maxid++,
+    obj,
+    x,
+    y
+  );
+  enigma::instance_alter = true;
```

Comment:
Is this ever read?

Reply:
prolly not

Reply:
Care to try removing it and find out?

Reply:
He removed it: e750a645ed49dca6124d20c72b0765f07dbe7f74

I haven't found evidence it causes any issues.